synapse/appservice/api.py

```diff
@@ -231,13 +231,32 @@ async def push_bulk(
                 json_body=body,
                 args={"access_token": service.hs_token},
             )
+            if logger.isEnabledFor(logging.DEBUG):
```

Comment:
I came back to this and was confused why I did this since it's guarding debug against debug.

AFAICT, this is to avoid potentially expensive string construction when we loop through all of the `events` here. We do this in a few other places in the codebase too.